barretenberg/cpp/src/barretenberg/vm2/simulation/bytecode_hashing.cpp

```diff
@@ -9,16 +9,17 @@ FF BytecodeHasher::compute_public_bytecode_commitment(const BytecodeId bytecode_
                                                       const std::vector<uint8_t>& bytecode)
 {
     auto bytecode_length_in_bytes = static_cast<uint32_t>(bytecode.size());
-    std::vector<FF> contract_bytecode_fields = encode_bytecode(bytecode);
-    FF running_hash = bytecode_length_in_bytes;
-    for (const FF& contract_bytecode_field : contract_bytecode_fields) {
-        // This emits events to our hasher (poseidon2 hash) subtrace
-        running_hash = hasher.hash({ contract_bytecode_field, running_hash });
-    }
+
+    std::vector<FF> inputs = { GENERATOR_INDEX__PUBLIC_BYTECODE };
+    auto bytecode_as_fields = encode_bytecode(bytecode);
+    inputs.insert(inputs.end(), bytecode_as_fields.begin(), bytecode_as_fields.end());
+
+    FF hash = hasher.hash(inputs);
+
     events.emit({ .bytecode_id = bytecode_id,
```

Comment:
you could comment this out (although cpp might complain about unused) in which case i would change 
`DefaultEventEmitter<BytecodeHashingEvent> bytecode_hashing_emitter;` to 
`NoopEventEmitter<BytecodeHashingEvent> bytecode_hashing_emitter;` in the `simulation_helper`